### DISCUSSION TITLE
Copilot code referencing in VSCode - Public Beta Feedback

### DISCUSSION THREAD

**Discussion by sharonlo** [community/community#77337] [Copilot Conversations]
We've launched Public Beta for Code referencing in VSCode! This feature searches across billions of files on public GitHub repositories for code that matches a Copilot suggestion. This is only available if you allow GitHub Copilot to suggest code completions that match publicly available code on GitHub.com. For more information, see "[Configuring GitHub Copilot settings on GitHub.com](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-settings-on-githubcom#enabling-or-disabling-duplication-detection)."

### What's changed since Private Beta?
Since [our initial launch in August](https://github.blog/2023-08-03-introducing-code-referencing-for-github-copilot/), we've heard your feedback around the noisiness of notifications and overall user flow. 

Now, if there's a match on a Copilot suggestion, you will find information about potential matches in the GitHub Copilot log. If you are interested in code references, you can refer to the window, otherwise, it won’t be in your way. 
<img src="https://github.com/community/community/assets/5484339/056455fb-560a-4f05-a041-0e503f71b517" width="600">

From there, if you want to learn more, a deep link will take you to a navigable page on GitHub.com to browse examples of the code match and their repository licenses, and see how many repositories — including ones without licenses — that code appears in, as well as links to those repositories.
<img src="https://github.com/community/community/assets/5484339/78ca0d1a-1fca-4c28-a114-7e419d872728" width="600">

Learn more about Copilot code referencing [here](https://docs.github.com/copilot/using-github-copilot/finding-public-code-that-matches-github-copilot-suggestions)

### We want to hear from you!

Feedback you'd like to share, any feature requests, we're all ears!

**Comment by pvys** [comment]
currently using vscode insider v1.85.0 with copilot v1.140.584, but I'm having some trouble getting it to work.
I enabled the public code setting and tried to verify it by following the fizz-buzz example, but the reference didn't appear in my log.
Also, I noticed that my output log is labeled 'Github Copilot' instead of 'Github Copilot Log.'
Are there any specific requirements I should be aware of to use this feature?

**Reply by sharonlo** [reply]
Hi @pvys, a couple things to troubleshoot. See if this works:

1. In your "GitHub Copilot" log do you see "Public code references are enabled"?
![Screenshot 2023-12-05 at 1 16 53 PM](https://github.com/community/community/assets/152307179/79bc0908-32d4-4432-a18f-a51ce10fb2e6)

2. Now, make sure you switch to a different dropdown to the "GitHub Copilot Log" not "GitHub Copilot" (they are different). In that log output, you should see references.
![Screenshot 2023-12-05 at 1 12 44 PM](https://github.com/community/community/assets/5484339/5f91e0f0-3fdc-4265-bfef-123287d85299)
![Screenshot 2023-12-05 at 1 17 16 PM](https://github.com/community/community/assets/152307179/3ca23d81-7f47-41d3-b19a-41d9fcffd868)

**Reply by pvys** [reply]
Sadly, this is what my screenshot looks like. I don't see 'GitHub Copilot Log' anywhere in the interface.
![dd](https://github.com/community/community/assets/55968684/010d0dda-8b89-4056-880f-68f83dfe2f5d)
As you can see in the screenshot, I've enabled public code suggestion.
![de](https://github.com/community/community/assets/55968684/6afddfb4-4ed7-4a39-b812-1eb96e2fd2f6)

**Reply by pvys** [reply]
I'm not certain why, but it has appeared today. I'll give it a try. Thanks!

**Reply by rahimbhuiyan2030** [reply]
How can registration the apps MoOver-pro https://private-user-images.githubusercontent.com/108199742/291005630-7c3627ea-2606-43a6-a366-956ac4286d9d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MTM1MDYsIm5iZiI6MTcwNDcxMzIwNiwicGF0aCI6Ii8xMDgxOTk3NDIvMjkxMDA1NjMwLTdjMzYyN2VhLTI2MDYtNDNhNi1hMzY2LTk1NmFjNDI4NmQ5ZC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwOFQxMTI2NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYmMzMmUzMDBmYjQ0ODZmYmFjM2FkZjM5Yzg3ODIzNGNkMzExMmZmMjZlMzA5Y2ZlMWM0OGFlZTk3MjUxZTQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.q9zXSGd_pBE034hHQO6gPh8bTy39VUGFoh5sov8O3EI

**Comment by davetapley** [comment]
Neat concept, but it just popped for the first time for me and it was..
.. unhelpful at best, and confusing until I figured out what's going on.

I forked https://github.com/textualize/textual/, and had been working for an hour or two when it popped up:
https://github.com/github-copilot/code_referencing?cursor=a7b344ed788ecaf413615ff645f25e19&editor=vscode

Of the 4 references:
* One is the upstream repo itself
* One is another fork
* Two are repos where people have (against best practices) checked in their Python virtual env


In all cases I presume it would be _reasonably_ easy to figure that this is the same code, either:
1. The matching code is in a repo which is an upstream fork (on GitHub) of the currently checked out code
2. It's a downstream fork (aka a sibling)
3. It's in a subdirectory of `site-packages` (or other language's conventional place for dependencies)

**Reply by sharonlo** [reply]
thanks @davetapley! That's great feedback, & we'll take this into account for our next iteration. We wanted to be over-inclusive of potential references, but noted how that's not helpful and duplicative.

**Reply by ONYEKA121** [reply]
Hi am Onyeka 
I like to learn help me to lean

**Comment by nhanvnguyen** [comment]
how to do that?

**Comment by rahimbhuiyan2030** [comment]
how can registration the apps and opening please i asking you i wating yours answering
![Screenshot_20231216_194740_One UI Home](https://github.com/community/community/assets/108199742/bf5e9b42-1221-42e0-ad26-dfb0c47dc28d)

**Comment by hazimin** [comment]
BETINA JALANG BANGSAT LEBIH BAGUS HANG PERGI MAMPUSSS CEPAT SIKIT

Pada Sel, 9 Jan 2024, 01:47 rahimbhuiyan2030 ***@***.***>
menulis:

**Comment by vsoto37** [comment]
> We've launched Public Beta for Code referencing in VSCode! This feature searches across billions of files on public GitHub repositories for code that matches a Copilot suggestion. This is only available if you allow GitHub Copilot to suggest code completions that match publicly available code on GitHub.com. For more information, see "[Configuring GitHub Copilot settings on GitHub.com](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-settings-on-githubcom#enabling-or-disabling-duplication-detection)."
> 
> ### What's changed since Private Beta?
> Since [our initial launch in August](https://github.blog/2023-08-03-introducing-code-referencing-for-github-copilot/), we've heard your feedback around the noisiness of notifications and overall user flow.
> 
> Now, if there's a match on a Copilot suggestion, you will find information about potential matches in the GitHub Copilot log. If you are interested in code references, you can refer to the window, otherwise, it won’t be in your way. ![](https://private-user-images.githubusercontent.com/5484339/287837079-056455fb-560a-4f05-a041-0e503f71b517.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MzMxNTMsIm5iZiI6MTcwNDgzMjg1MywicGF0aCI6Ii81NDg0MzM5LzI4NzgzNzA3OS0wNTY0NTVmYi01NjBhLTRmMDUtYTA0MS0wZTUwM2Y3MWI1MTcuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMjA0MDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTU5YTdiNGY5MTgzYTQ4YjEwOGRjNzFjYWExYzI0NDc2MTRhMjJjZWNlZGRlYzJjMmQ0MWE5ZWYyNjVlYzcxMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dxCsuucMSrAQQdt-uPqTGnQ6hOk8CjuMaD-P5_hOHc8) [ ![287837079-056455fb-560a-4f05-a041-0e503f71b517.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MzMxNTMsIm5iZiI6MTcwNDgzMjg1MywicGF0aCI6Ii81NDg0MzM5LzI4NzgzNzA3OS0wNTY0NTVmYi01NjBhLTRmMDUtYTA0MS0wZTUwM2Y3MWI1MTcuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMjA0MDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTU5YTdiNGY5MTgzYTQ4YjEwOGRjNzFjYWExYzI0NDc2MTRhMjJjZWNlZGRlYzJjMmQ0MWE5ZWYyNjVlYzcxMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dxCsuucMSrAQQdt-uPqTGnQ6hOk8CjuMaD-P5_hOHc8](https://private-user-images.githubusercontent.com/5484339/287837079-056455fb-560a-4f05-a041-0e503f71b517.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MzMxNTMsIm5iZiI6MTcwNDgzMjg1MywicGF0aCI6Ii81NDg0MzM5LzI4NzgzNzA3OS0wNTY0NTVmYi01NjBhLTRmMDUtYTA0MS0wZTUwM2Y3MWI1MTcuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMjA0MDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTU5YTdiNGY5MTgzYTQ4YjEwOGRjNzFjYWExYzI0NDc2MTRhMjJjZWNlZGRlYzJjMmQ0MWE5ZWYyNjVlYzcxMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dxCsuucMSrAQQdt-uPqTGnQ6hOk8CjuMaD-P5_hOHc8) ](https://private-user-images.githubusercontent.com/5484339/287837079-056455fb-560a-4f05-a041-0e503f71b517.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MzMxNTMsIm5iZiI6MTcwNDgzMjg1MywicGF0aCI6Ii81NDg0MzM5LzI4NzgzNzA3OS0wNTY0NTVmYi01NjBhLTRmMDUtYTA0MS0wZTUwM2Y3MWI1MTcuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMjA0MDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTU5YTdiNGY5MTgzYTQ4YjEwOGRjNzFjYWExYzI0NDc2MTRhMjJjZWNlZGRlYzJjMmQ0MWE5ZWYyNjVlYzcxMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dxCsuucMSrAQQdt-uPqTGnQ6hOk8CjuMaD-P5_hOHc8) [ ](https://private-user-images.githubusercontent.com/5484339/287837079-056455fb-560a-4f05-a041-0e503f71b517.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MzMxNTMsIm5iZiI6MTcwNDgzMjg1MywicGF0aCI6Ii81NDg0MzM5LzI4NzgzNzA3OS0wNTY0NTVmYi01NjBhLTRmMDUtYTA0MS0wZTUwM2Y3MWI1MTcuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMjA0MDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTU5YTdiNGY5MTgzYTQ4YjEwOGRjNzFjYWExYzI0NDc2MTRhMjJjZWNlZGRlYzJjMmQ0MWE5ZWYyNjVlYzcxMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dxCsuucMSrAQQdt-uPqTGnQ6hOk8CjuMaD-P5_hOHc8)
> 
> From there, if you want to learn more, a deep link will take you to a navigable page on GitHub.com to browse examples of the code match and their repository licenses, and see how many repositories — including ones without licenses — that code appears in, as well as links to those repositories. <img alt="" width="600" src="https://private-user-images.githubusercontent.com/5484339/287837098-78ca0d1a-1fca-4c28-a114-7e419d872728.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4MzMxNTMsIm5iZiI6MTcwNDgzMjg1MywicGF0aCI6Ii81NDg0MzM5LzI4NzgzNzA5OC03OGNhMGQxYS0xZmNhLTRjMjgtYTExNC03ZTQxOWQ4NzI3MjgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDlUMjA0MDUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2UyMDQ3MWZkZTdhNGE4MmVmNmQ1MmU2NTkwNTdkYTFjMjE4Y2ZlZGEyNmNmOTE0YWQxOTkzZTFhYjA1ZTZmOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ysA_VcuBOVw2YRt0z8B9tntndAT8JYv-UBzjZO4IEww">
> 
> Learn more about Copilot code referencing [here](https://docs.github.com/copilot/using-github-copilot/finding-public-code-that-matches-github-copilot-suggestions)
> 
> ### We want to hear from you!
> Feedback you'd like to share, any feature requests, we're all ears!
Request Id: 182c67d2-9492-4a7a-8ff7-c1fca34fb60b
Correlation Id: 90095cab-ae89-4e0f-9836-81a42e152fdb
Timestamp: 2024-01-09T20:26:05.024Z

**Comment by doruk5** [comment]
Yaragi yediniz

**Comment by hazimin** [comment]
It works!

BETINA JALANG BANGSAT LEBIH BAGUS HANG PERGI MAMPUSSS CEPAT SIKIT

Pada Jum, 19 Jan 2024, 7:12 PG doruk5 ***@***.***> menulis:

**Comment by JaKoutny** [comment]
Last week Code Reference in VS Code worked fine, but now I don't have "GitHub Copilot Log" available in Output Window. (Matching Public Code is allowed).
Has anything changed?
